### ISSUE TITLE
[YSQL] : WHERE condition option 32 not supported yet

### ISSUE THREAD

**Issue by kmuthukk** [YugaByte/yugabyte-db#1094]
@dbronzetti added (on slack) that:

<<< 
I’m doing the tests with the version 1.2.2.0 as you suggested but I’m getting the error at the beginning of my model replication:

```
ActiveRecord::StatementInvalid: PG::FeatureNotSupported: \
   ERROR:  WHERE condition option 32 not supported yet
DETAIL:  The WHERE condition option is not supported yet.
```
>>>

CC: @nocaway , @m-iancu .

_Originally posted by @kmuthukk in https://github.com/YugaByte/yugabyte-db/issues/1088#issuecomment-478032420_

**Comment by robertpang** [comment]
WHERE condition option 32 denotes the `<col> IN (1, 3, 5, ...)` condition. It is not supported yet in YSQL beta. As a workaround, replace the condition with `(<col> = 1 OR <col> = 3 OR <col> = 5 ...)` if feasible.

**Comment by dbronzetti** [comment]
@robertpang , was this because it was found something wrong with that condition?, because I made several tests with version 1.2.0.2 and that never happened on any of my tests. 

Not sure if you're aware we're running a rails app and this query is part of the postgresql adapter which is part of a gem from the app, and it didnt change at all in my tests.

**Comment by robertpang** [comment]
@dbronzetti In version 1.2.0.2, we did not handle the few WHERE condition options passed by Postgres to our YugaByte backend and inaccurate condition comparison results were returned as a result. In 1.2.2.0, we made a change to return an error for now when those operators like `IN` are used. It is in our TODO list to support those options quickly.

**Comment by dbronzetti** [comment]
@robertpang that explains why no error was returned on my previous tests. Thanks for clarifying this buddy. 

In order to continue helping you I did some changes on gem the query (just to keep moving with our model replication into Yugabye), so from this:

```
exec_query(<<-SQL, 'SCHEMA').rows.first[0].to_i > 0
SELECT COUNT(*)
FROM pg_class c
LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE c.relkind in ('v','r')
AND c.relname = '#{table.gsub(/(^"|"$)/,'')}'
AND n.nspname = #{schema ? "'#{schema}'" : 'ANY (current_schemas(false))'}
SQL
end
```
to your suggestion:
```
exec_query(<<-SQL, 'SCHEMA').rows.first[0].to_i > 0
SELECT COUNT(*)
FROM pg_class c
LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE (c.relkind = 'v' OR c.relkind = 'r')
AND c.relname = '#{table.gsub(/(^"|"$)/,'')}'
AND n.nspname = #{schema ? "'#{schema}'" : 'ANY (current_schemas(false))'}
SQL
end
```

but the same error was raised:
```
ActiveRecord::StatementInvalid: PG::FeatureNotSupported: ERROR:  WHERE condition option 32 not supported yet
DETAIL:  The WHERE condition option is not supported yet.
```
is this happening because of the `= ANY()` condition?

**Comment by robertpang** [comment]
Yes. `ANY()` is another form of WHERE condition with multiple values. We should be able to support those conditions very soon.

**Comment by robertpang** [comment]
Fix has been implemented in commit [be0908b](https://github.com/YugaByte/yugabyte-db/commit/be0908b45827c69ff010d9263ff865310ce722a1).
